---
title: "Script3_Compiling_RNAseq_Counts"
author: "Abby Deaven"
date: "2024-02-09"
output: html_document
---

This script was used to compile readcounts files generated by Subread. RNA-seq experiments were grouped into 9 batches and run in parallel. Readcount files were output into separate folders.

Example of file organization: 

./CountFiles > Batch1 > SRR###### > SRR######_counts.txt

Files were mapped in "batches" based on the experimental condition. The counts files from each batch are separated into different directories.


```{r setup}
knitr::opts_chunk$set(echo = TRUE)

BiocManager::install("scater")

install.packages("purrr")
install.packages("tidyverse")
install.packages("vctrs")
install.packages("ggcorrplot")
library(purrr)
library(tidyverse)
library(vctrs)
library(scater)
library(ggcorrplot)

#set working directory to the correct location for working machine
setwd("./Matrix_Plotting")

read_in_feature_counts<- function(file){
cnt<- read_tsv(file, col_names =T, comment = "#")
#cnt<- cnt %>% dplyr::select(-Chr, -Start, -End, -Strand, -Length)
return(cnt)
}

```

## First step is to construct matrixes for each batch, using this command to iterate through folders of count files:

```{r batch 1 - asexual development, message=FALSE}
### Read in all .counts files in a directory 

f_files_1<- list.files("./Matrix_Plotting/CountFiles/Batch1/", pattern = ".?_counts.txt", full.names = T, recursive = T)
f_files_1 <- grep(f_files_1, pattern='.?_counts.txt.summary', invert=TRUE, value=TRUE)

read_in_feature_counts<- function(file){
  cnt<- read_tsv(file, col_names =T, comment = "#")
  #cnt<- cnt %>% dplyr::select(-Chr, -Start, -End, -Strand, -Length)
  return(cnt)
}
   
#Creating count matrix of geneID and loci:
raw_counts_1<- map(f_files_1, read_in_feature_counts)
raw_counts_df_1<- purrr::reduce(raw_counts_1, inner_join) 

#move GeneID to row name
raw_counts_df_1<-column_to_rownames(raw_counts_df_1, var = "Geneid")

##need to clean up the column names by stripping off##
#"../MappingOutput/bamFiles/SRR10916379/SRR10916379_Aligned.sortedByCoord.out.bam"
colnames(raw_counts_df_1) <- gsub("./MappingOutput/bamFiles/", "", colnames(raw_counts_df_1))
colnames(raw_counts_df_1) <- gsub('/SRR[0-9]*_Aligned.sortedByCoord.out.bam', "", colnames(raw_counts_df_1))

#Write raw counts to a table
write.table(raw_counts_df_1, file="./Matrix_Plotting/CountMatrices/Batch1_Counts.txt", sep="\t", quote=FALSE, col.names=NA, row.names=TRUE)

#Average replicates

asexualdev_avg <- raw_counts_df_1 %>%
  rownames_to_column(var = "avg_GeneID") %>%
  rowwise() %>%
  mutate(avg_SRR13977073 = mean(c_across(c(SRR13977073, SRR13977074, SRR13977075)), na.rm = TRUE)) %>%
  mutate(avg_SRR13977076 = mean(c_across(c(SRR13977076, SRR13977077, SRR13977077)), na.rm = TRUE)) %>%
  mutate(avg_SRR13977079 = mean(c_across(c(SRR13977079, SRR13977080, SRR13977081)), na.rm = TRUE)) %>%
  mutate(avg_SRR13977082 = mean(c_across(c(SRR13977082, SRR13977083, SRR13977084)), na.rm = TRUE)) %>%
  mutate(avg_SRR13977086 = mean(c_across(c(SRR13977086, SRR13977087, SRR13977088)), na.rm = TRUE)) %>%
  mutate(avg_SRR13977089 = mean(c_across(c(SRR13977089, SRR13977090, SRR13977091)), na.rm = TRUE)) %>%
  mutate(avg_SRR13977092 = mean(c_across(c(SRR13977092, SRR13977093, SRR13977094)), na.rm = TRUE)) %>%
  mutate(avg_SRR13977095 = mean(c_across(c(SRR13977095, SRR13977096, SRR13977097)), na.rm = TRUE)) %>%
  mutate(avg_SRR5824367 = mean(c_across(c(SRR5824367, SRR5824368)), na.rm = TRUE)) %>%
  mutate(avg_SRR5824369 = mean(c_across(c(SRR5824369, SRR5824370, SRR5824371)), na.rm = TRUE)) %>%
  mutate(avg_SRR5824373 = mean(c_across(c(SRR5824373, SRR5824374)), na.rm = TRUE)) %>%
  mutate(avg_SRR5824375 = mean(c_across(c(SRR5824375, SRR5824376, SRR5824377)), na.rm = TRUE)) %>%
  mutate(avg_SRR5824379 = mean(c_across(c(SRR5824379, SRR5824380)), na.rm = TRUE)) %>%
  mutate(avg_SRR5824381 = mean(c_across(c(SRR5824381, SRR5824382, SRR5824383)), na.rm = TRUE)) %>%
  mutate(avg_SRR5824385 = mean(c_across(c(SRR5824385, SRR5824386)), na.rm = TRUE)) %>%
  mutate(avg_SRR5824387 = mean(c_across(c(SRR5824387, SRR5824388, SRR5824389)), na.rm = TRUE)) %>%
  mutate(avg_SRR5000481 = mean(c_across(c(SRR5000481, SRR5000482)), na.rm = TRUE))

asexualdev_avg_df <- asexualdev_avg %>% select(starts_with("avg")) %>%
     column_to_rownames(var="avg_GeneID")

  colnames(asexualdev_avg_df) = gsub("avg_", "", colnames(asexualdev_avg_df))

### Colnames of averaged samples = lowest SRR for each sample

write.table(asexualdev_avg, file="./MatrixPlotting/CountFiles/AsexualDev_AvgCounts.txt", sep="\t", quote=FALSE, col.names=NA, row.names=TRUE)
 

```

```{r batch 2 - carbon/nitrogen sources}
### Read in all .counts files in a directory 

f_files_2<- list.files("./Matrix_Plotting/CountFiles/Batch2/", pattern = ".?_counts.txt", full.names = T, recursive = T)
f_files_2 <- grep(f_files_2, pattern='.?_counts.txt.summary', invert=TRUE, value=TRUE)

   read_in_feature_counts<- function(file){
  cnt<- read_tsv(file, col_names =T, comment = "#")
  #cnt<- cnt %>% dplyr::select(-Chr, -Start, -End, -Strand, -Length)
  return(cnt)
}




#Creating count matrix of geneID and loci:
raw_counts_2<- map(f_files_2, read_in_feature_counts)
raw_counts_df_2<- purrr::reduce(raw_counts_2, inner_join) 

#move GeneID to row name
raw_counts_df_2<-column_to_rownames(raw_counts_df_2, var = "Geneid")

##need to clean up the column names by stripping off##
#"../MappingOutput/bamFiles/SRR10916379/SRR10916379_Aligned.sortedByCoord.out.bam"
colnames(raw_counts_df_2) <- gsub("./MappingOutput/bamFiles/", "", colnames(raw_counts_df_2))
colnames(raw_counts_df_2) <- gsub('/SRR[0-9]*_Aligned.sortedByCoord.out.bam', "", colnames(raw_counts_df_2))

write.table(raw_counts_df_2, file="./Matrix_Plotting/CountMatrices/Batch2_Counts.txt", sep="\t", quote=FALSE, col.names=NA, row.names=TRUE)


#average rows#
carbon_avg <- raw_counts_df_2 %>%
    rownames_to_column(var = "avg_GeneID") %>%
  rowwise() %>%
#  mutate(avg_SRR22543632 = mean(c_across(c(SRR22543632, SRR22543633, SRR22543634)), na.rm = TRUE)) %>%
  mutate(avg_SRR8860870 = mean(c_across(c(SRR8860870, SRR8860871)), na.rm = TRUE)) %>%
  mutate(avg_SRR8860872 = mean(c_across(c(SRR8860872, SRR8860873)), na.rm = TRUE)) %>%
  mutate(avg_SRR8860874 = mean(c_across(c(SRR8860874, SRR8860875)), na.rm = TRUE)) %>%
  mutate(avg_SRR8860876 = mean(c_across(c(SRR8860876, SRR8860877)), na.rm = TRUE)) %>%
  mutate(avg_SRR5058897 = mean(c_across(c(SRR5058897, SRR5058898, SRR5058899)), na.rm = TRUE)) %>%
  mutate(avg_SRR5058900 = mean(c_across(c(SRR5058900, SRR5058901, SRR5058902)), na.rm = TRUE)) %>%
  mutate(avg_SRR5058903 = mean(c_across(c(SRR5058903, SRR5058904, SRR5058905)), na.rm = TRUE)) %>%
  mutate(avg_SRR5058906 = mean(c_across(c(SRR5058906, SRR5058907, SRR5058908)), na.rm = TRUE)) %>%
  mutate(avg_SRR3207373 = mean(c_across(c(SRR3207373, SRR3207374)), na.rm = TRUE)) %>%
  mutate(avg_SRR3207375 = mean(c_across(c(SRR3207375, SRR3207376)), na.rm = TRUE)) %>%
  mutate(avg_SRR3207377 = mean(c_across(c(SRR3207377, SRR3207378)), na.rm = TRUE)) %>%
  mutate(avg_SRR3207379 = mean(c_across(c(SRR3207379, SRR3207380)), na.rm = TRUE)) %>%
  mutate(avg_SRR12553384 = mean(c_across(c(SRR12553384, SRR12553385)), na.rm = TRUE)) %>%
  mutate(avg_SRR12553386 = mean(c_across(c(SRR12553386, SRR12553387)), na.rm = TRUE)) %>%
  mutate(avg_SRR999602 = mean(c_across(c(SRR999602, SRR999604)), na.rm = TRUE)) %>%
  mutate(avg_SRR999605 = mean(c_across(c(SRR999605, SRR999606)), na.rm = TRUE)) %>%
  mutate(avg_SRR2007027 = mean(c_across(c(SRR2007027, SRR2007028, SRR2007029)), na.rm = TRUE)) %>%
  mutate(avg_SRR999605 = mean(c_across(c(SRR999605, SRR999606)), na.rm = TRUE)) %>%
  mutate(avg_SRR999607 = mean(c_across(c(SRR999607)), na.rm = TRUE)) %>%
  mutate(avg_SRR1562549 = sum(c_across(c(SRR1562549)), na.rm = TRUE)) %>%
  mutate(avg_SRR1562550 = sum(c_across(c(SRR1562550)), na.rm = TRUE)) %>%
  mutate(avg_SRR1562551 = sum(c_across(c(SRR1562551)), na.rm = TRUE)) %>%
  mutate(avg_SRR1562552 = sum(c_across(c(SRR1562552)), na.rm = TRUE)) %>%
  mutate(avg_SRR1562553 = sum(c_across(c(SRR1562553)), na.rm = TRUE)) %>%
  mutate(avg_SRR1562554 = sum(c_across(c(SRR1562554)), na.rm = TRUE))


 carbon_avg_df <- carbon_avg %>% select(starts_with("avg"))
  colnames(carbon_avg_df) = gsub("avg_", "", colnames(carbon_avg_df))
  
write.table(carbon_avg_df, file="./MatrixPlotting/Batch2_CarbonSource_Avg_Counts.txt", sep="\t", quote=FALSE, col.names=NA, row.names=TRUE)

nitrogen_avg <- raw_counts_df_2 %>%
  rownames_to_column(var = "avg_GeneID") %>%
  rowwise() %>%
  mutate(avg_SRR11768114 = mean(c_across(c(SRR11768114, SRR11768115, SRR11768116)), na.rm = TRUE)) %>%
  mutate(avg_SRR11768117 = mean(c_across(c(SRR11768117, SRR11768118, SRR11768119)), na.rm = TRUE)) %>%
  mutate(avg_SRR11768120 = mean(c_across(c(SRR11768121, SRR11768122, SRR11768120)), na.rm = TRUE)) %>%
  mutate(avg_SRR11768123 = mean(c_across(c(SRR11768123, SRR11768124, SRR11768125)), na.rm = TRUE)) %>%
  mutate(avg_SRR11768126 = mean(c_across(c(SRR11768126, SRR11768127, SRR11768128)), na.rm = TRUE)) %>%
  mutate(avg_SRR11768129 = mean(c_across(c(SRR11768129, SRR11768130, SRR11768131)), na.rm = TRUE)) %>%
  mutate(avg_SRR11768132 = mean(c_across(c(SRR11768132, SRR11768133, SRR11768134)), na.rm = TRUE)) %>%
  mutate(avg_SRR11768135 = mean(c_across(c(SRR11768135, SRR11768136, SRR11768137)), na.rm = TRUE)) %>%
  mutate(avg_SRR11768138 = mean(c_across(c(SRR11768138, SRR11768139, SRR11768140)), na.rm = TRUE)) %>%
  mutate(avg_SRR11768141 = mean(c_across(c(SRR11768141, SRR11768142, SRR11768143)), na.rm = TRUE)) %>%
  mutate(avg_SRR11768150 = mean(c_across(c(SRR11768150, SRR11768151, SRR11768152)), na.rm = TRUE))

nitrogen_avg_df <- nitrogen_avg %>% select(starts_with("avg")) %>%
  column_to_rownames(var="avg_GeneID")

  colnames(nitrogen_avg_df) = gsub("avg_", "", colnames(nitrogen_avg_df))

write.table(nitrogen_avg_df, file="./MatrixPlotting/Batch2_nitrogenSource_Avg_Counts.txt", sep="\t", quote=FALSE, col.names=NA, row.names=TRUE)

##Making one comprehensive table
carbon_nitrogen_source_avg_df <- carbon_avg_df %>% bind_cols(nitrogen_avg_df) %>%
  column_to_rownames(var = "GeneID")
  rownames(carbon_nitrogen_source_avg_df) = rownames(nitrogen_avg_df)
  rownames(carbon_nitrogen_source_avg_df) = rownames(nitrogen_avg_df)

  write.table(carbon_nitrogen_source_avg_df, file="./MatrixPlotting/CountFiles/CarbonNitrogen_AvgCounts.txt", sep="\t", quote=FALSE, col.names=NA, row.names=TRUE)



write.table(carbon_avg, file="./MatrixPlotting/CountFiles/CarbonNitrogen_AvgCounts.txt", sep="\t", quote=FALSE, col.names=NA, row.names=TRUE)


```


```{r batch 4 - controls, message=FALSE}
### Read in all .counts files in a directory 

f_files_4 <- list.files("./Matrix_Plotting/CountFiles/Batch4/", pattern = ".?_counts.txt", full.names = T, recursive = T)
f_files_4 <- grep(f_files_4, pattern='.?_counts.txt.summary', invert=TRUE, value=TRUE)

#Creating count matrix of geneID and loci:
raw_counts_4<- map(f_files_4, read_in_feature_counts)
raw_counts_df_4<- purrr::reduce(raw_counts_4, inner_join) 

#move GeneID to row name
raw_counts_df_4<-column_to_rownames(raw_counts_df_4, var = "Geneid")

##need to clean up the column names by stripping off##
#"../MappingOutput/bamFiles/SRR10916379/SRR10916379_Aligned.sortedByCoord.out.bam"
colnames(raw_counts_df_4) <- gsub("./MappingOutput/bamFiles/", "", colnames(raw_counts_df_4))
colnames(raw_counts_df_4) <- gsub('/SRR[0-9]*_Aligned.sortedByCoord.out.bam', "", colnames(raw_counts_df_4))

raw_counts_df_4_counts <- raw_counts_df_4[,6:48]

#plotting correlation to look for outliers
corr4 = round(cor(raw_counts_df_4_counts), 1)
p.mat4 <- cor_pmat(raw_counts_df_4_counts)

# using hierarchical clustering
CONTROL_corr <- ggcorrplot(corr4,
#           hc.order = TRUE, 
           type = "lower", 
#           outline.color = "white",
           p.mat = p.mat4,
  tl.cex = 15, 
  lab=TRUE, lab_size = 2,
  ggtheme = ggplot2::theme_minimal)

png("./Supplement/Controls_Correlation_Plot.png",width=30,height=30,units="cm",res=1200)
print(CONTROL_corr)
dev.off()


##Samples to remove based on clustering: 
##the following samples were excluded from the averaged dataset because they show poor clustering with other control samples: SRR12066473	SRR12066475	SRR12066477	SRR3207372	SRR5320489	SRR5320490	SRR13859509

#counts_char=as.character(raw_counts_df)
raw_counts_df_4_filtered <- raw_counts_df_4 %>% select(-SRR12066473,-SRR12066475,-SRR12066477,-SRR3207372,-SRR5320489,-SRR5320490,-SRR13859509)
write.table(raw_counts_df_4, file="./Matrix_Plotting/CountMatrices/Batch4_Counts.txt", sep="\t", quote=FALSE, col.names=NA, row.names=TRUE)

  write.table(raw_counts_df_4_filtered, file="./Matrix_Plotting/CountMatrices/Batch4_Counts_filtered.txt", sep="\t", quote=FALSE, col.names=NA, row.names=TRUE)


controls_avg <- raw_counts_df_4_filtered %>%
  rownames_to_column(var = "avg_GeneID") %>%
  rename("Length"="avg_Length") %>%
  rowwise() %>%
  mutate(avg_SRR17761892 = mean(c_across(c(SRR17761892, SRR17761893, SRR17761894)), na.rm = TRUE)) %>%
  mutate(avg_SRR5177529 = mean(c_across(c(SRR5177529, SRR5177530)), na.rm = TRUE)) %>%
  mutate(avg_SRR12066472 = mean(c_across(c(SRR12066472, SRR12066474, SRR12066476)), na.rm = TRUE)) %>%
#  mutate(avg_SRR12066473 = mean(c_across(c(SRR12066473, SRR12066475, SRR12066477)), na.rm = TRUE)) %>%
  mutate(avg_SRR3207371 = mean(c_across(c(SRR3207371)), na.rm = TRUE)) %>%
  mutate(avg_SRR23545778 = mean(c_across(c(SRR23545778, SRR23545779, SRR23545780)), na.rm = TRUE)) %>%
  mutate(avg_SRR11266659 = mean(c_across(c(SRR11266659, SRR11266660)), na.rm = TRUE)) %>%
  mutate(avg_SRR11806662 = mean(c_across(c(SRR11806662, SRR11806663, SRR11806664)), na.rm = TRUE)) %>%
  mutate(avg_SRR11806677 = mean(c_across(c(SRR11806677, SRR11806678, SRR11806679)), na.rm = TRUE)) %>%
  mutate(avg_SRR13067280 = mean(c_across(c(SRR13067280, SRR13067281, SRR13067282)), na.rm = TRUE)) %>%
  mutate(avg_SRR11768110 = mean(c_across(c(SRR11768110, SRR11768111, SRR11768112, SRR11768113)), na.rm = TRUE)) %>%
  mutate(avg_SRR13859512 = mean(c_across(c(SRR13859512, SRR13859513)), na.rm = TRUE)) %>%
  mutate(avg_SRR10509603 = mean(c_across(c(SRR10509603, SRR10509604)), na.rm = TRUE)) %>%
#  mutate(avg_SRR3623690 = mean(c_across(c(SRR3623690, SRR3623694)), na.rm = TRUE)) %>% ### replicates of SRR13859508 -- all cluster poorly with other controls
  mutate(avg_SRR797950 = mean(c_across(c(SRR797950)), na.rm = TRUE))


controls_avg_df <- controls_avg %>% select(starts_with("avg"))
rownames(controls_avg_df) = rownames(raw_counts_df_4)

colnames(controls_avg_df) = gsub("avg_", "", colnames(controls_avg_df))

write.table(controls_avg_df, file="./MatrixPlotting/CountFiles/Controls_AvgCounts.txt", sep="\t", quote=FALSE, col.names=NA, row.names=TRUE)


```

```{r batch 5 - light response}
### Read in all .counts files in a directory 

f_files_5<- list.files("./Matrix_Plotting/CountFiles/Batch5/", pattern = ".?_counts.txt", full.names = T, recursive = T)
f_files_5 <- grep(f_files_5, pattern='.?_counts.txt.summary', invert=TRUE, value=TRUE)

#Creating count matrix of geneID and loci:
raw_counts_5<- map(f_files_5, read_in_feature_counts)
raw_counts_df_5<- purrr::reduce(raw_counts_5, inner_join) 

#move GeneID to row name
raw_counts_df_5<-column_to_rownames(raw_counts_df_5, var = "Geneid")

##need to clean up the column names by stripping off##
#"../MappingOutput/bamFiles/SRR10916379/SRR10916379_Aligned.sortedByCoord.out.bam"
colnames(raw_counts_df_5) <- gsub("./MappingOutput/bamFiles/", "", colnames(raw_counts_df_5))
colnames(raw_counts_df_5) <- gsub('/SRR[0-9]*_Aligned.sortedByCoord.out.bam', "", colnames(raw_counts_df_5))

#counts_char=as.character(raw_counts_df)
write.table(raw_counts_df_5, file="./Matrix_Plotting/CountMatrices/Batch5_Counts.txt", sep="\t", quote=FALSE, col.names=NA, row.names=TRUE)

light_response_avg <- raw_counts_df_5 %>%
  rownames_to_column(var = "avg_GeneID") %>%
  rowwise() %>%
  mutate(avg_SRR1055985 = mean(c_across(c(SRR1055985, SRR1055990)), na.rm = TRUE)) %>%
  mutate(avg_SRR1055986 = mean(c_across(c(SRR1055986, SRR1055991)), na.rm = TRUE)) %>%
  mutate(avg_SRR1055987 = mean(c_across(c(SRR1055987, SRR1055992)), na.rm = TRUE)) %>%
  mutate(avg_SRR1055988 = mean(c_across(c(SRR1055988, SRR1055993)), na.rm = TRUE)) %>%
  mutate(avg_SRR1055989 = mean(c_across(c(SRR1055989, SRR1055994)), na.rm = TRUE)) %>%
  mutate(avg_SRR8065394 = mean(c_across(c(SRR8065394, SRR8065395, SRR8065396, SRR8065397)), na.rm = TRUE)) %>%
  mutate(avg_SRR8065398 = mean(c_across(c(SRR8065398, SRR8065399, SRR8065400, SRR8065401)), na.rm = TRUE)) %>%
  mutate(avg_SRR7082724 = sum(c_across(c(SRR7082724)), na.rm = TRUE)) %>%
  mutate(avg_SRR7082725 = sum(c_across(c(SRR7082725)), na.rm = TRUE)) %>%
  mutate(avg_SRR7082726 = sum(c_across(c(SRR7082726)), na.rm = TRUE)) %>%
  mutate(avg_SRR7082727 = sum(c_across(c(SRR7082727)), na.rm = TRUE)) %>%
  mutate(avg_SRR7082728 = sum(c_across(c(SRR7082728)), na.rm = TRUE)) %>%
  mutate(avg_SRR7082729 = sum(c_across(c(SRR7082729)), na.rm = TRUE)) %>%
  mutate(avg_SRR7082730 = sum(c_across(c(SRR7082730)), na.rm = TRUE)) %>%
  mutate(avg_SRR7082731 = sum(c_across(c(SRR7082731)), na.rm = TRUE)) %>%
  mutate(avg_SRR7082732 = sum(c_across(c(SRR7082732)), na.rm = TRUE)) %>%
  mutate(avg_SRR7082733 = sum(c_across(c(SRR7082733)), na.rm = TRUE)) %>%
  mutate(avg_SRR7082734 = sum(c_across(c(SRR7082734)), na.rm = TRUE)) %>%
  mutate(avg_SRR7082735 = sum(c_across(c(SRR7082735)), na.rm = TRUE)) %>%
  ungroup()

light_response_avg_df <- light_response_avg %>% select(starts_with("avg")) %>%
  column_to_rownames(var = "avg_GeneID")

colnames(light_response_avg_df) = gsub("avg_", "", colnames(light_response_avg_df))


write.table(light_response_avg_df, file="./MatrixPlotting/CountFiles/LightResponse_AvgCounts.txt", sep="\t", quote=FALSE, col.names=NA, row.names=TRUE)

```


```{r batch 7 - sexual development, message=FALSE, warning=FALSE}
### Read in all .counts files in a directory 

f_files_7<- list.files("./Matrix_Plotting/CountFiles/Batch7/", pattern = ".?_counts.txt", full.names = T, recursive = T)
f_files_7 <- grep(f_files_7, pattern='.?_counts.txt.summary', invert=TRUE, value=TRUE)

#Creating count matrix of geneID and loci:
raw_counts_7<- map(f_files_7, read_in_feature_counts)
raw_counts_df_7<- purrr::reduce(raw_counts_7, inner_join) 

#move GeneID to row name
raw_counts_df_7<-column_to_rownames(raw_counts_df_7, var = "Geneid")

##need to clean up the column names by stripping off##
#"../MappingOutput/bamFiles/SRR10916379/SRR10916379_Aligned.sortedByCoord.out.bam"
colnames(raw_counts_df_7) <- gsub("./MappingOutput/bamFiles/", "", colnames(raw_counts_df_7))
colnames(raw_counts_df_7) <- gsub('/SRR[0-9]*_Aligned.sortedByCoord.out.bam', "", colnames(raw_counts_df_7))

#counts_char=as.character(raw_counts_df)
write.table(raw_counts_df_7, file="./Matrix_Plotting/CountMatrices/Batch7_Counts.txt", sep="\t", quote=FALSE, col.names=NA, row.names=TRUE)

raw_counts_df_7_mat <- raw_counts_df_7 %>%
  select(-Chr, -Start, -End, -Strand) %>%
  mutate_all(as.numeric)

c7 = cor(raw_counts_df_7_mat)
corrplot(c7, method = 'color')


sex_dev_avg <- raw_counts_df_7 %>%
      rownames_to_column(var = "avg_GeneID") %>%
rowwise() %>%
mutate(avg_SRR5177521 = mean(c_across(c(SRR5177521, SRR5177522)), na.rm = TRUE)) %>%
mutate(avg_SRR5177523 = mean(c_across(c(SRR5177523, SRR5177524)), na.rm = TRUE)) %>%
mutate(avg_SRR5177525 = mean(c_across(c(SRR5177525, SRR5177526)), na.rm = TRUE)) %>%
mutate(avg_SRR5177527 = mean(c_across(c(SRR5177527, SRR5177528)), na.rm = TRUE)) %>%
ungroup()

sex_dev_avg_df <- sex_dev_avg %>%
  select(starts_with("avg")) %>%
  column_to_rownames(var =  "avg_GeneID")


sexdev2 <- raw_counts_df_7_mat %>%
select(starts_with("SRR5856")) 
all_sexdev_avg_df <- sex_dev_avg_df %>% bind_cols(sexdev2)


colnames(all_sexdev_avg_df) = gsub("avg_", "", colnames(all_sexdev_avg_df))



write.table(all_sexdev_avg_df, file="./MatrixPlotting/CountFiles/SexDev_AvgCounts.txt", sep="\t", quote=FALSE, col.names=NA, row.names=TRUE)

```


```{r batch 8 - Stress/Drug Response, message=FALSE, warning=TRUE}
### Read in all .counts files in a directory 

f_files_8<- list.files("./Matrix_Plotting/CountFiles/Batch8/", pattern = ".?_counts.txt", full.names = T, recursive = T)
f_files_8 <- grep(f_files_8, pattern='.?_counts.txt.summary', invert=TRUE, value=TRUE)

#Creating count matrix of geneID and loci:
raw_counts_8<- map(f_files_8, read_in_feature_counts)
raw_counts_df_8<- purrr::reduce(raw_counts_8, inner_join) 

#move GeneID to row name
raw_counts_df_8<-column_to_rownames(raw_counts_df_8, var = "Geneid")

##need to clean up the column names by stripping off##
#"../MappingOutput/bamFiles/SRR10916379/SRR10916379_Aligned.sortedByCoord.out.bam"
colnames(raw_counts_df_8) <- gsub("./MappingOutput/bamFiles/", "", colnames(raw_counts_df_8))
colnames(raw_counts_df_8) <- gsub('/SRR[0-9]*_Aligned.sortedByCoord.out.bam', "", colnames(raw_counts_df_8))

#counts_char=as.character(raw_counts_df)
write.table(raw_counts_df_8, file="./Matrix_Plotting/CountMatrices/Batch8_Counts.txt", sep="\t", quote=FALSE, col.names=NA, row.names=TRUE)


stress_response_avg <- raw_counts_df_8 %>%
  rownames_to_column(var = "avg_GeneID") %>%
  rowwise() %>%
  mutate(avg_SRR1043827 = mean(c_across(c(SRR1043827, SRR1043828, SRR1043829)), na.rm = TRUE)) %>%
  mutate(avg_SRR1043830 = sum(c_across(c(SRR1043830)), na.rm = TRUE)) %>%
  mutate(avg_SRR1043831 = sum(c_across(c(SRR1043831)), na.rm = TRUE)) %>%
  mutate(avg_SRR1043832 = sum(c_across(c(SRR1043832)), na.rm = TRUE)) %>%
  mutate(avg_SRR1043833 = sum(c_across(c(SRR1043833)), na.rm = TRUE)) %>%
  mutate(avg_SRR1043834 = sum(c_across(c(SRR1043834)), na.rm = TRUE)) %>%
  mutate(avg_SRR2952642 = sum(c_across(c(SRR2952642)), na.rm = TRUE)) %>%
  mutate(avg_SRR2952643 = sum(c_across(c(SRR2952643)), na.rm = TRUE)) %>%
  mutate(avg_SRR2952644 = sum(c_across(c(SRR2952644)), na.rm = TRUE)) %>%
  mutate(avg_SRR5000483 = mean(c_across(c(SRR5000483, SRR5000484)), na.rm = TRUE)) %>%
  mutate(avg_SRR5000485 = mean(c_across(c(SRR5000485, SRR5000486)), na.rm = TRUE)) %>%
  mutate(avg_SRR1025973 = mean(c_across(c(SRR1025973, SRR1025974, SRR1025975)), na.rm = TRUE)) %>%
  mutate(avg_SRR1043589 = mean(c_across(c(SRR1043589, SRR1043590, SRR1043591)), na.rm = TRUE)) %>%
  mutate(avg_SRR1043593 = mean(c_across(c(SRR1043593, SRR1043594, SRR1043595)), na.rm = TRUE)) %>%
  mutate(avg_SRR1043597 = mean(c_across(c(SRR1043597, SRR1043598, SRR1043599)), na.rm = TRUE)) %>%
  mutate(avg_SRR1043592 = mean(c_across(c(SRR1043592, SRR1043596, SRR1043600)), na.rm = TRUE)) %>%
#  mutate(avg_SRR22543626 = mean(c_across(c(SRR22543626, SRR22543627, SRR22543628)), na.rm = TRUE)) %>%
  mutate(avg_SRR17761883 = mean(c_across(c(SRR17761883, SRR17761884, SRR17761885)), na.rm = TRUE)) %>%
  mutate(avg_SRR17761886 = mean(c_across(c(SRR17761886, SRR17761887, SRR17761888)), na.rm = TRUE)) %>%
  mutate(avg_SRR17761889 = mean(c_across(c(SRR17761889, SRR17761890, SRR17761891)), na.rm = TRUE)) %>%
  mutate(avg_SRR12066483 = mean(c_across(c(SRR12066483, SRR12066485, SRR12066487)), na.rm = TRUE)) %>%
  mutate(avg_SRR12066484 = mean(c_across(c(SRR12066484, SRR12066486, SRR12066487)), na.rm = TRUE)) %>%
  mutate(avg_SRR3207363 = mean(c_across(c(SRR3207363, SRR3207364)), na.rm = TRUE)) %>%
  mutate(avg_SRR3207367 = sum(c_across(c(SRR3207367)), na.rm = TRUE)) %>%
  mutate(avg_SRR3207369= sum(c_across(c(SRR3207369)), na.rm = TRUE)) %>%
  mutate(avg_SRR3207370= sum(c_across(c(SRR3207370)), na.rm = TRUE)) %>%
  mutate(avg_SRR1593998 = mean(c_across(c(SRR1593998, SRR1594003, SRR1594011)), na.rm = TRUE)) %>%
  mutate(avg_SRR1593999 = sum(c_across(c(SRR1593999)), na.rm = TRUE)) %>%
  mutate(avg_SRR1594001 = sum(c_across(c(SRR1594001)), na.rm = TRUE)) %>%
  mutate(avg_SRR1594002 = sum(c_across(c(SRR1594002)), na.rm = TRUE)) %>%
  mutate(avg_SRR1594000 = mean(c_across(c(SRR1594000, SRR1594004, SRR1594012)), na.rm = TRUE)) %>%
  mutate(avg_SRR23545772 = mean(c_across(c(SRR23545772, SRR23545773, SRR23545774)), na.rm = TRUE)) %>%
  mutate(avg_SRR12553380 = mean(c_across(c(SRR12553380, SRR12553381)), na.rm = TRUE)) %>%
  mutate(avg_SRR12553382 = mean(c_across(c(SRR12553382, SRR12553383)), na.rm = TRUE)) %>%
  mutate(avg_SRR11768144 = mean(c_across(c(SRR11768144, SRR11768145, SRR11768146, SRR11768147, SRR11768148, SRR11768149)), na.rm = TRUE)) %>%
  mutate(avg_SRR11768153 = mean(c_across(c(SRR11768153, SRR11768154, SRR11768155)), na.rm = TRUE)) %>%
  mutate(avg_SRR14428135 = mean(c_across(c(SRR14428135, SRR14428136, SRR14428137)), na.rm = TRUE)) %>%
  mutate(avg_SRR14428141 = mean(c_across(c(SRR14428141, SRR14428142, SRR14428143)), na.rm = TRUE)) %>%
  ungroup()

stress_response_avg <- as.data.frame(stress_response_avg)
stress_response_avg_df <- stress_response_avg %>% 
  select(starts_with("avg")) %>% 
  column_to_rownames(var = "avg_GeneID")
    
    
    
#rownames(stress_response_avg_df) = rownames(raw_counts_df_8)
colnames(stress_response_avg_df) = gsub("avg_", "", colnames(stress_response_avg_df))

write.table(stress_response_avg_df, file="./MatrixPlotting/CountFiles/StressResponse_AvgCounts.txt", sep="\t", quote=FALSE, col.names=NA, row.names=TRUE)

```

```{r batch 9 - wild isolates}

### Read in all .counts files in a directory 

f_files_9<- list.files("./Matrix_Plotting/CountFiles/Batch9/", pattern = ".?_counts.txt", full.names = T, recursive = T)
f_files_9 <- grep(f_files_9, pattern='.?_counts.txt.summary', invert=TRUE, value=TRUE)

#Creating count matrix of geneID and loci:
raw_counts_9<- map(f_files_9, read_in_feature_counts)
raw_counts_df_9<- purrr::reduce(raw_counts_9, inner_join) 

#move GeneID to row name
raw_counts_df_9<-column_to_rownames(raw_counts_df_9, var = "Geneid")

##need to clean up the column names by stripping off##
#"../MappingOutput/bamFiles/SRR10916379/SRR10916379_Aligned.sortedByCoord.out.bam"
colnames(raw_counts_df_9) <- gsub("./MappingOutput/bamFiles/", "", colnames(raw_counts_df_9))
colnames(raw_counts_df_9) <- gsub('/SRR[0-9]*_Aligned.sortedByCoord.out.bam', "", colnames(raw_counts_df_9))

#counts_char=as.character(raw_counts_df)
write.table(raw_counts_df_9, file="./MatrixPlotting/CountFiles/WildIsolates_Counts.txt", sep="\t", quote=FALSE, col.names=NA, row.names=TRUE)

wildisolates_countsonly <- raw_counts_df_9 %>%
  select(-Length, -Chr, -Start, -End, -Strand) %>%
  mutate_all(as.numeric)


## no averages to condense ###
```

Make master raw count database
```{r combining master count/TPM matrix}
    wildisolates_countsonly <- read.table("./MatrixPlotting/CountFiles/WildIsolates_Counts.txt", sep = "\t", header = TRUE)
    light_response_avg_df <- read.table("./MatrixPlotting/CountFiles/LightResponse_AvgCounts.txt", sep = "\t", header = TRUE)
    carbon_nitrogen_source_avg_df <- read.table("./MatrixPlotting/CountFiles/CarbonNitrogen_AvgCounts.txt", sep = "\t", header = TRUE) 
    asexualdev_avg_df <- read.table("./MatrixPlotting/CountFiles/AsexualDev_AvgCounts.txt", sep = "\t", header = TRUE) 
    all_sexdev_avg_df <- read.table("./MatrixPlotting/CountFiles/SexDev_AvgCounts.txt", sep = "\t", header = TRUE) 
    stress_response_avg_df <- read.table("./MatrixPlotting/CountFiles/StressResponse_AvgCounts.txt", sep = "\t", header = TRUE) 

##making masterlist of counts
master_counts_df <- raw_counts_df_4_filtered %>%
#relocate(GeneID, before = everything()) %>%
  bind_cols(select(wildisolates_countsonly, starts_with("SRR"))) %>%
  bind_cols(select(raw_counts_df_8, starts_with("SRR"))) %>%
  bind_cols(select(raw_counts_df_5, starts_with("SRR"))) %>%
  bind_cols(select(raw_counts_df_2, starts_with("SRR"))) %>%
  bind_cols(select(raw_counts_df_1, starts_with("SRR"))) %>%
  bind_cols(select(raw_counts_df_7, starts_with("SRR"))) %>%
  column_to_rownames(var = "GeneID")

    
##Make master list of averaged counts    
avg_master_counts_df <- controls_avg_df %>%
#relocate(GeneID, before = everything()) %>%
  bind_cols(select(stress_response_avg_df, starts_with("SRR"))) %>%
  bind_cols(select(wildisolates_countsonly, starts_with("SRR"))) %>%
  bind_cols(select(light_response_avg_df, starts_with("SRR"))) %>%
  bind_cols(select(carbon_nitrogen_source_avg_df, starts_with("SRR"))) %>%
  bind_cols(select(asexualdev_avg_df, starts_with("SRR"))) %>%
  bind_cols(select(all_sexdev_avg_df, starts_with("SRR"))) %>%
  column_to_rownames(var = "GeneID")


colnames(master_counts_df)[1] <- "Length"

write.table(master_counts_df, file="./MatrixPlotting/Master_Counts.txt", sep="\t", quote=FALSE, col.names=NA, row.names=TRUE)

write.table(avg_master_counts_df, file="./MatrixPlotting/Average_Master_Counts.txt", sep="\t", quote=FALSE, col.names=NA, row.names=TRUE)

###Convert to TPMs##
all_tpm <- calculateTPM(avg_master_counts_df[,2:285], lengths = avg_master_counts_df$Length)

write.table(all_tpm, file="./MatrixPlotting/Average_Master_TPM.txt", sep="\t", quote=FALSE)


```

Compiling counts for generated RNA-seq experiments

```{r false perithecia RNA-seq}

### Read in all .counts files in a directory 

f_files_fp<- list.files("./Datasets/FalsePeritheciaRNAseq/", pattern = ".?_counts.txt", full.names = T, recursive = T)
f_files_fp <- grep(f_files_fp, pattern='.?_counts.txt.summary', invert=TRUE, value=TRUE)




read_in_feature_counts<- function(file){
  cnt<- read_tsv(file, col_names =T, comment = "#")
  #cnt<- cnt %>% dplyr::select(-Chr, -Start, -End, -Strand, -Length)
  return(cnt)
}
   
#Creating count matrix of geneID and loci:
raw_counts_fp<- map(f_files_fp, read_in_feature_counts)

raw_counts_fp_df<- as.data.frame(purrr::reduce(raw_counts_fp, inner_join))

#move GeneID to row name
raw_counts_fp_df<-column_to_rownames(raw_counts_fp_df, var = "Geneid")

##need to clean up the column names by stripping off##
#"../MappingOutput/bamFiles/SRR10916379/SRR10916379_Aligned.sortedByCoord.out.bam"

#Files have slightly different names
colnames(raw_counts_fp_df) <- gsub(".*/bamFiles/","", colnames(raw_counts_fp_df))
colnames(raw_counts_fp_df) <- gsub('-[1-9]*_cDNA', "", colnames(raw_counts_fp_df))
colnames(raw_counts_fp_df) <- gsub('*Aligned.sortedByCoord.out.bam', "", colnames(raw_counts_fp_df))
#colnames(raw_counts_df) <- gsub('', "", colnames(raw_counts_df))


countsonly_fp<- raw_counts_fp_df[,6:ncol(raw_counts_fp_df)]
raw_counts_fp_df$Length <- as.numeric(raw_counts_fp_df[,5])

#mastercounts2 <- countsonly_fp[rownames(raw_counts_df) == "NCU06402",]

write.table(raw_counts_fp_df, file = "./MatrixPlotting/FP_counts.txt", sep = "\t")

fp_tpm <- calculateTPM(countsonly_fp, lengths = raw_counts_fp_df$Length)

write.table(fp_tpm, file = "./MatrixPlotting/FP_tpm.txt", sep = "\t")


##merging samples that are sequencing duplicates
mergedcounts <- cbind(fp_countsonly[,1:3], rowMeans(fp_countsonly[,c(4,15)]), rowMeans(fp_countsonly[,c(5,16)]), rowMeans(fp_countsonly[,c(6,17)]), rowMeans(fp_countsonly[,c(7,18)]), rowMeans(fp_countsonly[,c(8,19)]), rowMeans(fp_countsonly[,c(9,20)]), rowMeans(fp_countsonly[,c(10,21)]), rowMeans(fp_countsonly[,c(11,22)]), rowMeans(fp_countsonly[,c(12,23)]), rowMeans(fp_countsonly[,c(13,24)]), rowMeans(fp_countsonly[,c(14,25)]))

colnames(mergedcounts) = c("set7_Mycelia_Rep1","set7_Mycelia_Rep2", "set7_Mycelia_Rep3", "set7_matAA_FP_Rep1", "set7_matAA_FP_Rep2", "set7_matAA_FP_Rep3", "set7_mata_FP_Rep1", "set7_mata_FP_Rep2", "WT_matAA_PP_Rep1", "WT_matAA_PP_Rep2", "WT_matAA_PP_Rep3", "WT_mata_PP_Rep1", "WT_mata_PP_Rep2", "WT_mata_PP_Rep3")

write.table(mergedcounts, file="./MatrixPlotting/FP_Counts_Merged.txt", sep ="\t")

## Averaging samples
avg_mergedcounts <- cbind(rowMeans(mergedcounts[,1:3], na.rm = TRUE), rowMeans(mergedcounts[,4:6] , na.rm = TRUE), rowMeans(mergedcounts[,7:8] , na.rm = TRUE), rowMeans(mergedcounts[,9:11]), rowMeans(mergedcounts[,12:14]))
colnames(avg_mergedcounts) <- c("set7_Mycelia", "set7_matAA_FP","set7_mata_FP","WT_matAA_PP","WT_mata_PP")

fp_tpm <- data.frame(calculateTPM(mergedcounts, lengths = raw_counts_df$Length))
fp_avg_tpm <- data.frame(calculateTPM(avg_mergedcounts, lengths = raw_counts_df$Length))
fp_tpm <- fp_tpm %>% rownames_to_column(var = "GeneID")


write.table(raw_counts_fp_df, file = "./MatrixPlotting/FP_counts.txt", sep = "\t")

write.table(fp_tpm, file = "./MatrixPlotting/FP_merge_tpm.txt", sep = "\t")

write.table(fp_avg_tpm, file = "./MatrixPlotting/FP_avg_tpm.txt", sep = "\t")

```

